### DISCUSSION TITLE
Aplicação da Calculadora quando debug da erro de execução e permissão.

### DISCUSSION THREAD

**Discussion by nanslack** [balta-io/forum#457] [Tira Dúvidas] [answered]
Oi bom dia estou com erro em aplicativo da calculadora acho que por ter trocado a versão do .Net pode estar pedindo alguma função pode me ajudar 
segue o Código:
using System;
namespace calculadora
{
    public class Program()
    {
        static void Main(string[] args)
        {
            Menu();
        }
        static void Soma()
        {
            Console.Clear();
            Console.WriteLine("Calculadora Simples:");

            Console.WriteLine("entre com o primeiro valor");
            float v1 = float.Parse(Console.ReadLine());
            Console.WriteLine("entre com o segundo valor");
            float v2 = float.Parse(Console.ReadLine());
            float result = v1 + v2;
            Console.WriteLine($"o valor da soma dos dois numeros é {result}");
            Console.ReadKey();
            Menu();
        }
        static void Subtracao()
        {
            Console.Clear();
            Console.WriteLine("Calculadora Simples:");

            Console.WriteLine("entre com o primeiro valor");
            float v1 = float.Parse(Console.ReadLine());
            Console.WriteLine("entre com o segundo valor");
            float v2 = float.Parse(Console.ReadLine());
            float result = v1 - v2;
            Console.WriteLine($"o valor da subtração dos dois numeros é {result}");
            Console.ReadKey();
            Menu();
        }
        static void Divisao()
        {
            Console.Clear();
            Console.WriteLine("Calculadora Simples:");

            Console.WriteLine("entre com o primeiro valor");
            float v1 = float.Parse(Console.ReadLine());
            Console.WriteLine("entre com o segundo valor");
            float v2 = float.Parse(Console.ReadLine());
            float result = v1 / v2;
            Console.WriteLine($"o valor da divisão dos dois numeros é {result}");
            Console.ReadKey();
            Menu();
        }
        static void Multiplicação()
        {
            Console.Clear();
            Console.WriteLine("Calculadora Simples:");

            Console.WriteLine("entre com o primeiro valor");
            float v1 = float.Parse(Console.ReadLine());
            Console.WriteLine("entre com o segundo valor");
            float v2 = float.Parse(Console.ReadLine());
            float result = v1 * v2;
            Console.WriteLine($"o valor da multiplicação dos dois numeros é {result}");
            Console.ReadKey();
            Menu();
        }
        static void Expoente()
        {
            Console.Clear();
            Console.WriteLine("Entre com o valor de base");
            float v1 = float.Parse(Console.ReadLine());
            Console.WriteLine("Entre com o valor do expoente");
            float v2 = float.Parse(Console.ReadLine());
            double res = Math.Pow(v1, v2);
            Console.WriteLine($"O valor exponencial da operação é:{res}");
            Console.ReadKey();
            Menu();
        }
        static void Raiz()
        {
            Console.Clear();
            Console.WriteLine("Entre com o valor:");
            double valor = double.Parse(Console.ReadLine());
            double res = Math.Sqrt(valor);
            Console.WriteLine($"Ovalor da raiz quadrado do numero é:{res}");
            Console.ReadKey();
            Menu();
        }
        static void Menu()
        {
            Console.Clear();
            Console.WriteLine("Por favor escolha a função desejada");
            Console.WriteLine("1 - Somar");
            Console.WriteLine("2 - Subtrair");
            Console.WriteLine("3 - Dividir");
            Console.WriteLine("4 - Multiplicar");
            Console.WriteLine("5 - Exponencial de um numero");
            Console.WriteLine("6 - Raiz Quadrada de um numero:");
            Console.WriteLine("7 - Sair");
            Console.WriteLine("---------------------");

            short res = short.Parse(Console.ReadLine());
            switch (res)
            {
                case 1: Soma(); break;
                case 2: Subtracao(); break;
                case 3: Divisao(); break;
                case 4: Multiplicação(); break;
                case 5: Expoente(); break;
                case 6: Raiz(); break;
                case 7: System.Environment.Exit(0); break; // chamando a função sair e encerrar prorama
                default: Menu(); break; // chamando menu novamente caso o usuário não escolha o numero correto
            }
        }
    }
}

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @nanslack, como vai?

Qual erro está tendo, e em qual linha?

**Reply by nanslack** [reply]
![Captura de tela 2024-01-09 092745](https://github.com/balta-io/forum/assets/146471278/426f8933-7996-47d4-a9a8-fe1daf0e5bd7)
esse erro acontece quando compilo o código

**Reply by nanslack** [reply]
outra duvida agora que o .net core está na versão 8 que estou usando o code avisa que o código mudou para o nível diferente e não precisa mais do método Main()

**Reply by andrebaltieri** [reply]
@nanslack bom dia! Você pode por favor parar a depuração, fechar o VS Code e abrir novamente.
Em seguida, no terminal, digita os seguintes comandos:
```
dotnet clean
dotnet build
```

Depois tenta executar a aplicação novamente, com o comando `dotnet run`.

Sobre a mudança no escopo principal do programa, ela veio no .NET 6 mas só afeta a estrutura do Program, o código continua o mesmo.

Se você quiser manter a estrutura igual do curso, basta usar a flag `--use-program-main` na hora de criar o projeto:
```
dotnet new console -o MeuProjetoComDotNet --use-program-main
```

**Reply by nanslack** [reply]
Oi André então eu fiz os procedimentos anteriores e mesmo erro a execução da aplicação com dotnet run funciona normal o erro é quando eu compilo o programa ele está pedindo algum parâmetro
![Captura de tela 2024-01-09 111722](https://github.com/balta-io/forum/assets/146471278/5be25949-58c7-405e-91f3-c0096bd4b7ff)

**Reply by nanslack** [reply]
![Captura de tela 2024-01-09 111821](https://github.com/balta-io/forum/assets/146471278/1db82bc5-5d2e-49ba-ab03-db79ef537c5d)

**Reply by nanslack** [reply]
![Captura de tela 2024-01-09 112432](https://github.com/balta-io/forum/assets/146471278/042d9aeb-ee4d-4cac-82ea-eb4aaccbe64d)

**Reply by andrebaltieri** [reply]
Consegue deletar a pasta `.vscode` (Fecha o VS Code e faz isso pelo Windows) e depois abrir e tentar debugar novamente...
